array_linked_list.py/arrays.py

```diff
@@ -2,63 +2,103 @@
 
 # Do not use any of the built in array functions for this exercise
 class array:
-    def __init__(self):
+    def __init__(self, capacity):
+
         # Your code here
-        pass
+        self.capacity = capacity
+        self.elements = [None] * capacity
+        self.count = 0
 
 
 # Double the size of the given array
-def resize_array():
+def resize_array(array):
     # Your code here
-    pass
+    new_capacity = array.capacity * 2
+    new_elements = [None] * new_capacity
+
+    for i in range(array.capacity):
+        new_elements[i] = array.elements[i]
+
+    array.elements = new_elements
+    array.capacity = new_capacity
 
 
 # Return an element of a given array at a given index
-def array_read():
+def array_read(array, index):
     # Throw an error if array is out of the current count
     # Your code here
-    pass
+    if index >= array.count:
+        print(f'Error! index ' + str(index) + f'out of range')
```

Comment:
Another way to write this:

```
print(f'Error! index {index} out of range')
```

Reply:
Yea I wrote this from the Class Lecture the way that Artem was writing it. I did get a chance to go back and change it. But yes I like this way better thank you!